Commiting AWS Batch Lab (#132)
aws-samples/aws-hpc-tutorials: content/05-AWS Batch/02-build-docker-container.md

+++
title = "c. Build A Container With Docker"
date = 2019-09-18T10:46:30-04:00
weight = 30
tags = ["tutorial", "install", "AWS", "batch", "Docker"]
+++

> In this step, you will create a simple "Dockerfile" with the specifications for a container and then use Docker commands to build and test a container image on your Cloud9 instance. 

Reviewer: ```suggestion
In this step, you will create a **Dockerfile** file that contains the steps to build and test a container image on your Cloud9 instance. 
```

> 1. Open a new file in the Cloud9 editor named "Dockerfile" and cut and paste the following contents:

Reviewer: ```suggestion
1. In the Cloud9, create a new file in the Cloud9 editor named **Dockerfile** and execute the following command:
```

```bash

> FROM public.ecr.aws/amazonlinux/amazonlinux:latest

Reviewer: ```suggestion
cat > Dockerfile << EOF
FROM public.ecr.aws/amazonlinux/amazonlinux:latest
```

RUN yum -y update
RUN amazon-linux-extras install epel -y
RUN yum -y install stress-ng
RUN echo $'#!/bin/bash\n\
echo "Passing the following arguments to stress-ng: $STRESS_ARGS"\n\
/usr/bin/stress-ng $STRESS_ARGS' \n\ >> /docker-entrypoint.sh 
RUN chmod 0744 /docker-entrypoint.sh

> RUN cat /docker-entrypoint.sh

Reviewer: ```suggestion
RUN cat /docker-entrypoint.sh
EOF
```

ENTRYPOINT ["/docker-entrypoint.sh"]
```
This Docker file uses the latest version of Amazon Linux 2, installs the stress-ng executable from the EPEL repository, and creates a wrapper script called docker-entrypoint.sh that is executed when the container starts. This script uses an environment variable, $STRESS_ARGS, to pass command-line options to the stress-ng executable.

> 1. Build a container with the tag "stress-ng" from your Dockerfile by excuting the following command:

Reviewer: ```suggestion
2. Build a container with the tag "stress-ng" from your Dockerfile by executing the following command in the Cloud9 terminal:
```
